hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java

```diff
@@ -666,7 +667,8 @@ public void importContainerData(InputStream input,
 
   public void importContainerData(KeyValueContainerData originalContainerData)
       throws IOException {
-    containerData.setState(originalContainerData.getState());
+    // place the container in the Recovering state while it is being imported
+    containerData.setState(RECOVERING);
```

Comment:
This will not affect the container state on disk.
We need to change the container state before we do an atomic move and keep the original state of the container in memory.
https://github.com/apache/ozone/blob/7f11b7b55980024f65cfb5450070597a03fcfbee/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/TarContainerPacker.java#L102